l10n_it_ricevute_bancarie/views/configuration_view.xml

```diff
@@ -32,28 +32,68 @@
                     <separator colspan="4" string="Acceptance" />
                     <field
                         name="acceptance_journal_id"
+                        domain="[('type', '=', 'bank'),('company_id', '=', company_id)]"
```

Comment:
I vari domini aggiunti su `type` e `internal_type` non mi sembrano avere a che fare con la problematica in oggetto, si possono rimuovere o spostare in un commit dedicato?
Sulla loro correttezza mi rimetto al parere dei funzionali :innocent:.

Inoltre, parere mio, se questi domini devono valere sempre per questo modello allora li definirei nel model.

Reply:
Ho dovuto aggiungere i domini su `type` e `internal_type` anche nella vista perchè impostando solamente `('company_id', '=', company_id)` non teneva conto di quello impostato nel modello, nella selezione venivano tolti solo i record di altre aziende:
https://github.com/OCA/l10n-italy/blob/f549619d01be6c9c4e8a6ddcce98f2b393e905d6/l10n_it_ricevute_bancarie/models/riba_config.py#L61

Mi sbaglierò, ma non dovrebbe essere possibile mettere il dominio `('company_id', '=', company_id)` nella definizione del campo nel modello perchè il valore di `company_id` ce l'hai solo frontend.
Non so come spiegarmi meglio, ma anche usando i vari `lambda` nel dominio python non funzionava, però accetto consigli.

Reply:
Forse basta `check_company=True` come indicavi sopra

Reply:
@SirAionTech qualche idea per risolvere questo commento?

Reply:
> Ho dovuto aggiungere i domini su `type` e `internal_type` anche nella vista perchè impostando solamente `('company_id', '=', company_id)` non teneva conto di quello impostato nel modello, nella selezione venivano tolti solo i record di altre aziende:
> 
> https://github.com/OCA/l10n-italy/blob/f549619d01be6c9c4e8a6ddcce98f2b393e905d6/l10n_it_ricevute_bancarie/models/riba_config.py#L61
> 

Non ho capito: dici che per risolvere i problemi di conti multi-company è necessario aggiungere i domini su `type` e `internal_type`?
Per risolvere i problemi di multi-company penso siano sufficienti le modifiche riguardanti `company_id`.
Se vengono mostrati anche conti di tipi diversi è un altro problema, indipendente dai problemi di multi-company, che quindi metterei in un commit diverso e andrebbe tracciato a parte; anche perché c'è da valutare che i tipi selezionati siano corretti.

Unendo le soluzioni a due diversi problemi nello stesso commit/PR, il rischio è che se una delle due soluzioni deve essere rivista, rallenta il merge dell'altra.

> Mi sbaglierò, ma non dovrebbe essere possibile mettere il dominio `('company_id', '=', company_id)` nella definizione del campo nel modello perchè il valore di `company_id` ce l'hai solo frontend. Non so come spiegarmi meglio, ma anche usando i vari `lambda` nel dominio python non funzionava, però accetto consigli.

Vedi come è fatto in https://github.com/odoo/odoo/blob/551bf83f792813535cb571b8d746238d5fea6b50/addons/account/models/account_journal.py#L204

Reply:
> Non ho capito: dici che per risolvere i problemi di conti multi-company è necessario aggiungere i domini su `type` e `internal_type`? Per risolvere i problemi di multi-company penso siano sufficienti le modifiche riguardanti `company_id`. Se vengono mostrati anche conti di tipi diversi è un altro problema, indipendente dai problemi di multi-company, che quindi metterei in un commit diverso e andrebbe tracciato a parte; anche perché c'è da valutare che i tipi selezionati siano corretti.
Avevo dovuto forzarli aggiungendoli nel XML perchè se impostavo quello dell'azienda nella vista, nelle dropdown considerava solo quello "dimenticandosi" di quello definito nel modello python.
Con la soluzione sotto dovrei risolvere tutto.

> Vedi come è fatto in https://github.com/odoo/odoo/blob/551bf83f792813535cb571b8d746238d5fea6b50/addons/account/models/account_journal.py#L204
Ottimo, è quello che cercavo, grazie 👍

Reply:
:+1:

> se impostavo quello dell'azienda nella vista, nelle dropdown considerava solo quello "dimenticandosi" di quello definito nel modello python.

non se ne "dimenticava": il `domain` definito sulla vista sovrascrive il `domain` definito lato codice.